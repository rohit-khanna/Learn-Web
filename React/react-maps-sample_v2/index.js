import React from "react";
import { render } from "react-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "./index.css";
import App from "./src/components/App";
import ConfigureStore from "./src/redux/configureStore";
import InitialState from "./src/redux/reducers/InitialState";
import { Provider as ReduxProvider } from "react-redux";

const store = new ConfigureStore(InitialState);

render(
  <ReduxProvider store={store}>
    <App />
  </ReduxProvider>,
  document.getElementById("app")
);
