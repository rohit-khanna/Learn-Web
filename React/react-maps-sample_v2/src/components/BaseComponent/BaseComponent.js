import React, { Component } from "react";
import FallbackComponent from "../FallbackComponent";

/**
 * HOC: Base component. It Adds Error Handling capabilities and shows a fallback component
 * in case of error
 * @param {*} AppComponent Component
 */
const BaseComponent = AppComponent => {
  return class InnerBase extends Component {
    //contructor
    constructor(props) {
      super(props);
      this.state = { hasError: false, error: null, errorInfo: "" };
    }

    /**
	* Life cycle method called when exception is caught
	 * @param {*} error eoor details
	 * @param {*} errorInfo error stack trace
	 */
    componentDidCatch(error, errorInfo) {
      this.setState({
        error: error && error.message,
        errorInfo: errorInfo,
        hasError: true
      });
    }

    render() {
      const { error, hasError, errorInfo } = this.state;
      if (hasError)
        return <FallbackComponent errorInfo={errorInfo} error={error} />;
      return <AppComponent {...this.props} />;
    }
  };
};

export default BaseComponent;
