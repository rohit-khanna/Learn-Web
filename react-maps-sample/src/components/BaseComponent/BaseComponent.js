import React, { Component } from 'react';
import FallbackComponent from '../FallbackComponent';

const BaseComponent = (AppComponent) => {
	return class InnerBase extends Component {
		constructor(props) {
			super(props);
			this.state = { hasError: false, error: null, errorInfo: '' };
		}

		componentDidCatch(error, errorInfo) {
			this.setState({
				error: error && error.message,
				errorInfo: errorInfo,
				hasError: true
			});
		}

		render() {
			const { error, hasError, errorInfo } = this.state;
			if (hasError) return <FallbackComponent errorInfo={errorInfo} error={error} />;
			return <AppComponent {...this.props} />;
		}
	};
};

export default BaseComponent;
