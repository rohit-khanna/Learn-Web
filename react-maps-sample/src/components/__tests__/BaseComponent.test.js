import React from "react";
import BaseComponent from "../BaseComponent";
import renderer from "react-test-renderer";

test("BaseComponent Tests", () => {
  const component = renderer.create(BaseComponent(<div />));
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
