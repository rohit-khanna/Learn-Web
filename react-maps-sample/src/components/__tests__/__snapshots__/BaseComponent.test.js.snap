// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseComponent Tests 1`] = `null`;
